client/v12/openapi/api_ldap_identity_providers.go

```diff
@@ -0,0 +1,1318 @@
+
+	localVarHeaderParams := make(map[string]string)
+	localVarQueryParams := _neturl.Values{}
+	localVarFormParams := _neturl.Values{}
+
+	if r.authorization == nil {
+		return localVarReturnValue, nil, reportError("authorization is required and must be specified")
+	}
+
+	if r.query != nil {
+		localVarQueryParams.Add("query", parameterToString(*r.query, ""))
+	}
+	if r.range_ != nil {
+		localVarQueryParams.Add("range", parameterToString(*r.range_, ""))
+	}
+	if r.orderBy != nil {
+		localVarQueryParams.Add("orderBy", parameterToString(*r.orderBy, ""))
+	}
+	if r.descending != nil {
+		localVarQueryParams.Add("descending", parameterToString(*r.descending, ""))
+	}
+	// to determine the Content-Type header
+	localVarHTTPContentTypes := []string{}
+
+	// set Content-Type header
+	localVarHTTPContentType := selectHeaderContentType(localVarHTTPContentTypes)
+	if localVarHTTPContentType != "" {
+		localVarHeaderParams["Content-Type"] = localVarHTTPContentType
+	}
+
+	// to determine the Accept header
+	localVarHTTPHeaderAccepts := []string{"application/json"}
+
+	// set Accept header
+	localVarHTTPHeaderAccept := selectHeaderAccept(localVarHTTPHeaderAccepts)
+	if localVarHTTPHeaderAccept != "" {
+		localVarHeaderParams["Accept"] = localVarHTTPHeaderAccept
+	}
+	localVarHeaderParams["Authorization"] = parameterToString(*r.authorization, "")
+	req, err := r.apiService.client.prepareRequest(r.ctx, localVarPath, localVarHTTPMethod, localVarPostBody, localVarHeaderParams, localVarQueryParams, localVarFormParams, localVarFormFileName, localVarFileName, localVarFileBytes)
+	if err != nil {
+		return localVarReturnValue, nil, err
+	}
+
+	localVarHTTPResponse, err := r.apiService.client.callAPI(req)
+	if err != nil || localVarHTTPResponse == nil {
+		return localVarReturnValue, localVarHTTPResponse, err
+	}
+
+	localVarBody, err := _ioutil.ReadAll(localVarHTTPResponse.Body)
+	localVarHTTPResponse.Body.Close()
+	if err != nil {
+		return localVarReturnValue, localVarHTTPResponse, err
+	}
+
+	if localVarHTTPResponse.StatusCode >= 300 {
+		newErr := GenericOpenAPIError{
+			body:  localVarBody,
+			error: localVarHTTPResponse.Status,
+		}
+		if localVarHTTPResponse.StatusCode == 200 {
+			var v LdapIdentityProviderList
+			err = r.apiService.client.decode(&v, localVarBody, localVarHTTPResponse.Header.Get("Content-Type"))
+			if err != nil {
+				newErr.error = err.Error()
+				return localVarReturnValue, localVarHTTPResponse, newErr
+			}
+			newErr.model = v
+			return localVarReturnValue, localVarHTTPResponse, newErr
+		}
+		if localVarHTTPResponse.StatusCode == 401 {
+			var v Error
+			err = r.apiService.client.decode(&v, localVarBody, localVarHTTPResponse.Header.Get("Content-Type"))
+			if err != nil {
+				newErr.error = err.Error()
+				return localVarReturnValue, localVarHTTPResponse, newErr
+			}
+			newErr.model = v
+			return localVarReturnValue, localVarHTTPResponse, newErr
+		}
+		if localVarHTTPResponse.StatusCode == 403 {
```

Comment:
Is this auto-generated code? Else it could avoid some repetition as it seems we do the same thing for error codes 200, 401, 403 and 500.

I guess my question is more, do we expect this to be maintained manually or will the openapi code generator eventually support discriminators?

If this will be maintained manually we can avoid a lot of the code repetition.

Reply:
> Is this auto-generated code? Else it could avoid some repetition as it seems we do the same thing for error codes 200, 401, 403 and 500.
> 
> I guess my question is more, do we expect this to be maintained manually or will the openapi code generator eventually support discriminators?
> 
> If this will be maintained manually we can avoid a lot of the code repetition.


The `api_ldap_identity_providers.go` is a copy paste version of `api_identity_providers.go` with search and replace on the model  type

for example:
https://github.com/appgate/sdp-terraform-provider/blob/dcc54c745e55de253c5a2fc4c1cb2351e83a85fe/client/v12/openapi/api_identity_providers.go#L577
to return the correct model, model_ldap_provider.go `LdapProvider` instead of `IdentityProvider`

So yes, its right now manually maintained. Im not really happy with it either, but I found it to be the path of least resistance.

Reply:
I see, makes sense.